app/src/main/java/org/mozilla/fenix/nimbus/NimbusExperimentsFragment.kt

```diff
@@ -5,81 +5,119 @@
 
 /**
  * Fragment use for managing Nimbus experiments.
  */
 @Suppress("TooGenericExceptionCaught")
-class NimbusExperimentsFragment : Fragment(R.layout.mozac_service_nimbus_experiments) {
-
-    private var adapter: NimbusExperimentAdapter? = null
-
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        bindRecyclerView(view)
-    }
+class NimbusExperimentsFragment : Fragment() {
 
     override fun onResume() {
         super.onResume()
         showToolbar(getString(R.string.preferences_nimbus_experiments))
     }
 
-    override fun onDestroyView() {
-        super.onDestroyView()
-        // Letting go of the resources to avoid memory leak.
-        adapter = null
-    }
-
-    private fun bindRecyclerView(view: View) {
-        val experimentsView = NimbusExperimentsView(
-            navController = findNavController(),
-        )
-
-        val recyclerView = view.findViewById<RecyclerView>(R.id.nimbus_experiments_list)
-        recyclerView.layoutManager = LinearLayoutManager(requireContext())
-
-        val shouldRefresh = adapter != null
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View = ComposeView(requireContext()).apply {
 
-        lifecycleScope.launch(IO) {
-            try {
+        setContent {
+            FirefoxTheme {
                 val experiments =
                     requireContext().components.analytics.experiments.getAvailableExperiments()
 
-                lifecycleScope.launch(Main) {
-                    runIfFragmentIsAttached {
-                        if (!shouldRefresh) {
-                            adapter = NimbusExperimentAdapter(
-                                experimentsView,
-                                experiments,
+                NimbusExperiments(
+                    availableExperiments = experiments,
+                    onSelectedExperiment = { selectedExperiment ->
+                        val directions =
+                            NimbusExperimentsFragmentDirections.actionNimbusExperimentsFragmentToNimbusBranchesFragment(
+                                experimentId = selectedExperiment.slug,
+                                experimentName = selectedExperiment.userFacingName,
                             )
-                        }
 
-                        view.findViewById<TextView>(R.id.nimbus_experiments_empty_message).isVisible =
-                            experiments.isEmpty()
-                        recyclerView.adapter = adapter
+                        findNavController().navigate(directions)
                     }
+                )
+            }
+        }
+    }
+
+    /**
+     * The list for the Nimbus Experiments,
+     *
+     * @param availableExperiments List of available items
+     * @param onSelectedExperiment Callback for when item is selected.
```

Comment:
I will have further suggestions for improving the kDoc for your Composable. However, since I am already asking for a lot of changes from the get go, I would prefer if you could take some time to look at our existing @Composable and try to emulate the same style of how we write our kDoc.